### ISSUE TITLE
[a11y]: Group/Fieldset does not have an accessible name

### ISSUE THREAD

**Issue by Shimpali-Deshpande** [carbon-design-system/carbon#20503]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.70.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

During an accessibility review, we noticed that the TileGroup component triggers a "Group/Fieldset does not have an accessible name" warning in the IBM Equal Access Accessibility Checker.

We have screen that loads a Tearsheet on top of a Modal. As such 2 legends show up causing conflicts for Accessibility. Hence we added this temporary workaround to pass a visually hidden legend, which resolves the warning for now.

<img width="2990" height="1724" alt="Image" src="https://github.com/user-attachments/assets/9614870c-718e-4bea-8fcf-a9896e4da099" />

### Reproduction/example

-

### Steps to reproduce

Create a Tearsheet that has a button to open a Modal on top of it. Each will display a reusable TileSet with its own legend.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by alisonjoseph** [comment]
Hi, thanks for opening this, can you reproduce this in a [Stackblitz](https://stackblitz.com/github/carbon-design-system/sandboxes/tree/main/react/latest/react-19-vite/?preset=node=)?

**Comment by tay1orjones** [comment]
Closing due to no reproduction
